packages/@uppy/companion/src/server/controllers/callback.js

```diff
@@ -1,8 +1,24 @@
 /**
  * oAuth callback.  Encrypts the access token and sends the new token with the response,
  */
+const serialize = require('serialize-javascript')
+
 const tokenService = require('../helpers/jwt')
 const logger = require('../logger')
+const oAuthState = require('../helpers/oauth-state')
+
+const closePageHtml = (origin) => `
+  <!DOCTYPE html>
+  <html>
+  <head>
+      <meta charset="utf-8" />
+      <script>
+      window.opener.postMessage(${serialize({ error: true })}, ${serialize(origin)})
```

Comment:
We may want to add a check here.
```suggestion
      window.opener && window.opener.postMessage(${serialize({ error: true })}, ${serialize(origin)})
```
Or we could leave it as is, and if it throws, the window doesn't close, the users sees the `Authentication failed.` in the body and that's maybe better. In that case a comment could be useful:
```suggestion
      ` +
      // if window.opener is nullish, we want the following line to throw to avoid
      // the window closing without informing the user.
      `window.opener.postMessage(${serialize({ error: true })}, ${serialize(origin)})
```